src/hooks/core/push-notification/usePushNotificationHook.ts

```diff
@@ -0,0 +1,218 @@
+import PushNotification from 'react-native-push-notification';
+import PushNotificationIOS from '@react-native-community/push-notification-ios';
+import messaging from '@react-native-firebase/messaging';
+import React, {useState} from 'react';
+import {Platform} from 'react-native';
+import {useNavigation} from '@react-navigation/core';
+import {useRecoilState} from 'recoil';
+
+import ChannelList from '../../../database/schema/ChannelListSchema';
+import profileAtom from '../../../atom/profileAtom';
+import useChatUtilsHook from '../chat/useChatUtilsHook';
+import useLocalDatabaseHook from '../../../database/hooks/useLocalDatabaseHook';
+import useUserAuthHook from '../auth/useUserAuthHook';
+import TokenStorage, {ITokenEnum} from '../../../utils/storage/custom/tokenStorage';
+import {getAnonymousUserId, getUserId} from '../../../utils/users';
+
+const usePushNotificationHook = () => {
+  const isIos = Platform.OS === 'ios';
+
+  const navigation = useNavigation();
+  const {signedProfileId} = useUserAuthHook();
+  const {localDb} = useLocalDatabaseHook();
+  const {fetchChannelDetail, setSelectedChannel} = useChatUtilsHook();
+  const [, setProfileAtom] = useRecoilState(profileAtom);
+
+  const [isLoadingFetchingChannelDetail, setIsLoadingFetchingChannelDetail] = useState(false);
+
+  const __createChannel = () => {
+    PushNotification.createChannel(
+      {
+        channelId: 'bettersosialid', // (required)
+        channelName: 'New Messages & Comments', // (required)
+        playSound: true, // (optional) default: true
+        soundName: 'default', // (optional) See `soundName` parameter of `localNotification` function
+        importance: 4, // (optional) default: 4. Int value of the Android notification importance
+        vibrate: true // (optional) default: true. Creates the default vibration patten if true.
+      },
+      (created) => {
+        if (__DEV__) {
+          console.log(`createChannel returned '${created}'`);
+        }
+      } // (optional) callback returns whether the channel was created, false means it already existed.
+    );
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `__createChannel` creates a notification channel for Android. It's important to note that this function will only work on Android 8.0 (API level 26) and above, as notification channels were introduced in this version. For lower versions, this function will not have any effect. If you need to support lower versions, you should add a check for the Android version before calling this function.



<!-- This is an auto-generated comment by OSS CodeRabbit -->